src/main/scala/org/broadinstitute/dsde/firecloud/service/TrialService.scala

```diff
@@ -253,4 +259,16 @@ final class TrialService
   private def projectReport: Future[PerRequestMessage] =
     Future(RequestComplete(OK, trialDAO.projectReport))
 
+  private def recordUserAgreement(userInfo: UserInfo): Future[PerRequestMessage] = {
+    thurloeDao.getTrialStatus(userInfo.id, userInfo) flatMap {
+      case Some(status) => status.state match {
+        case Some(TrialStates.Enabled) => thurloeDao.saveTrialStatus(userInfo.id, userInfo, status.copy(userAgreed = true)) flatMap {
+          case Success(_) => Future(RequestComplete(NoContent))
+          case Failure(ex) => Future(RequestComplete(InternalServerError, ex.getMessage))
+        }
+        case _ => Future(RequestCompleteWithErrorReport(BadRequest, "You are not Enabled to agree to terms."))
```

Comment:
I think we should keep things like their trial status (`Enabled`) opaque from end users, it's not like there's anything they can do about it

Reply:
I wanted to distinguish between the error cases for developer purposes but I agree with your statement.